src/ts/components/fullscreentogglebutton.ts

```diff
@@ -28,17 +28,34 @@ export class FullscreenToggleButton extends ToggleButton<ToggleButtonConfig> {
       }
     };
 
+    let fullscreenAvailableHandler = () => {
+      if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
+        this.show();
+      } else {
+        this.hide();
+      }
+    };
+
+    uimanager.getConfig().events.onUpdated.subscribe(fullscreenAvailableHandler);
+
     player.on(player.exports.PlayerEvent.ViewModeChanged, fullscreenStateHandler);
 
     this.onClick.subscribe(() => {
-      if (player.getViewMode() === player.exports.ViewMode.Fullscreen) {
-        player.setViewMode(player.exports.ViewMode.Inline);
+      if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
+        if (player.getViewMode() === player.exports.ViewMode.Fullscreen) {
+          player.setViewMode(player.exports.ViewMode.Inline);
+        } else {
+          player.setViewMode(player.exports.ViewMode.Fullscreen);
+        }
       } else {
-        player.setViewMode(player.exports.ViewMode.Fullscreen);
+        if (console) {
+          console.log('Fullscreen unavailable');
+        }
       }
```

Comment:
With our current player code style, I'd collapse that `if` a little bit to get rid of the nested `if` s:

```ts
if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
  const targetViewMode =
    player.getViewMode() === player.exports.ViewMode.Fullscreen
      ? player.exports.ViewMode.Inline
      : player.exports.ViewMode.Fullscreen;
  
  player.setViewMode(targetViewMode);
} else if (console) {
  console.log('Fullscreen unavailable');
}
```

Reply:
Or what about an early return 🤔 Not sure what the code-style says about that 🤔 

ATM: this is just consistent with our code / naming style what we have e.g. in the `PictureInPictureToggleButton`

Reply:
kk. Sure, an early return would also be possible. Is just a suggestion anyway, not a change request 😉

Reply:
Sure I totally agree that those nested `if`s don't read nicely.

Reply:
251a0cd